src/scripts/main.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      e.preventDefault();
+
+      // eslint-disable-next-line no-undef
+      resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+    },
+    { once: true });
```

Comment:
It is not a good practice to prevent default event handling inside a generic function like this. The 'waitFor' function is supposed to be reusable and not tied to specific event handling logic. It's better to move 'e.preventDefault();' to the place where you actually handle the event. In this case, it would be better to place it inside the 'printMessage' function.